test/uriManagerSpec.js

```diff
@@ -87,7 +88,7 @@ describe('UriManager', function () {
       const self = this
       expect(function () {
         self.uri.parse('http://localhost/base/foo/abracadabra', 'example')
-      }).to.throw(Error, /URI is not a valid example URI/)
+      }).to.throw(InvalidUriParameterError, /URI is not a valid example URI/)
```

Comment:
It's more of an `InvalidUriError` (rather than `InvalidUriParameterError`). It could happen, for example, if the user tries to PUT an account with a transfer id as the `id`, in which case it would be an invalid request. It could also happen if the client method (in the ledger), erroneously calls `uri.parse(<account_uri>, 'transfer')`, in which case it would be some kind of server/API error. It's difficult to distinguish between the two cases without more granular error handling.

The correct errors depend on the consumer of the API. If it's the five-bells-* modules, we should be catching these errors and re-throwing as HTTP errors. If it's the user making the HTTP requests, we should be throwing HTTP errors from here. We probably want to support the first use case, long term. We're providing the second as a stop-gap solution, since the work required to support the first is fairly extensive.

With this in mind, I think it's OK to throw this as an invalid request error, since this is precisely the use case we want to support here. This ticket is motivated by the fact that the error is currently returned as a 500 with no message.